### PR TITLE
Specify number of cores for jump and ramp_fit

### PR DESCRIPTION
**Pull request by mwregan2**
This PR addresses the problem that for multiprocessing the number of cores to use can only be a fraction of the total number of cores. This is fine for small machines but on large clusters even 1/4 can be larger than someone is allowed to use. 
The PR adds the flexibility to specify explicitly the number of cores the user wants to use. Changes were made for both Jump and Ramp fitting and new methods were written with unit tests. 
No code changes are needed in the JWST repo only in the documentation.

**Checklist**
- [x] added entry in `CHANGES.rst` (either in `Bug Fixes` or `Changes to API`)
- [X] updated relevant tests
- [ ] updated relevant documentation
- [ ] updated relevant milestone(s)
- [x] added relevant label(s)

### CODE REVIEW

File: tests/test_ramp_fitting.py [hunk 1]
```diff
@@ -1567,3 +1568,20 @@ def print_all_info(slopes, cube, optional):
     print_slopes(slopes)
     print_integ(cube)
     print_optional(optional)
+
+def test_compute_num_slices():
```
Comment (kmacdonald-stsci) [inline tests/test_ramp_fitting.py:1572]: This test should be moved to line 1358.  The test file is divided into different sections.  The bottom section is for debugging functions.  There is a test suite section that currently ends at 1358.

**Review by kmacdonald-stsci — APPROVED** [review]
I'd like to request the ramp fit test be moved to a different location per my specific comment above.  Other than that, everything looks good.

File: src/stcal/ramp_fitting/ols_fit.py [hunk 2]
```diff
@@ -73,7 +74,8 @@ def ols_ramp_fit_multi(
 
     # Determine number of slices to use for multi-processor computations
     nrows = ramp_data.data.shape[2]
-    number_slices = utils.compute_slices(max_cores, nrows)
+    num_available_cores = cpu_count()
+    number_slices = utils.compute_num_slices(max_cores, nrows, num_available_cores)
```
[the remark this answers is not in the document]
Comment (mwregan2) [reply src/stcal/ramp_fitting/ols_fit.py:78]: Well, I tried to change the util.py version to be more specific to distinguish it from a similar named routine.
I had the same thought that a single routine would be better but I don't know how to do it. I tried to change the util.py routine as little as possible which ended up with the input parameters being in a different order. Anyway, that's easy to change.

[the remark this answers is not in the document]
Comment (hbushouse) [reply src/stcal/ramp_fitting/ols_fit.py:78]: OK, we'll just go with the two different functions for now.

**Comment by hbushouse** [comment]
OK, this is looking better. Just need to address the 2 errors coming from the code style CI test (removed unused imports from test modules).

**Comment by hbushouse** [comment]
Still need to have the style check errors fixed before we can merge:

tests/test_jump.py:3:24: F401 [*] `astropy.io.fits` imported but unused
tests/test_twopoint_difference.py:3:24: F401 [*] `astropy.io.fits` imported but unused

**Review by hbushouse — APPROVED** [review]
Finally all looks good.

**Pull request merged**